infra/azure/bash/ensure-resource-groups.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script creates a Azure resource group, storage accounts and storage needed
+# to store Terraform states.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# shellcheck disable=SC2034
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+
+if ! [[ -x "$(command -v jq)" ]]; then
+  echo "Please install jq. https://stedolan.github.io/jq/download/.  Aborting."
+  exit 1
+fi
+
+if ! [[ -x "$(command -v az)" ]]; then
+  echo "Please install Azure CLI. https://learn.microsoft.com/en-us/cli/azure/?view=azure-cli-latest.  Aborting."
+  exit 1
+fi
+
+RESOURCE_GROUP="k8s-infra-tf-states-rg"
+RESOURCE_GROUP_LOCATION=$(az group show --name $RESOURCE_GROUP --output tsv --query 'location')
+
+# storage accounts rules: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftstorage
+#TODO define a role assignement with Azure Entra Groups
+readonly TERRAFORM_STATE_BUCKET_ENTRIES=(
+  k8sinfratfstatesub
+  k8sinfratfstateprowbuild
```

Comment:
A single bucket is fine, given at users will be able to access all the buckets in the sub.

Reply:
You apply RBAC to the resource groups instead of the subscriptions.

Reply:
so one bucket where we keep tf changes impacting the subscription and  only accessible for the sub owners would be fine ?